### PR TITLE
Issue 2648 consent on metadata

### LINKED ISSUE #2648: ConsentInterceptor is called on metadata calls
Currently, consent interceptor is invoked during calls to /metadata, and the $meta operation. This is an error, and the consent interceptor should ignore checks on those endpoints. 

Solution: 
1. if a request is a meta operation, or navigating to /metadata, the hook should just return
2. We should still call `completeOperationFailure()` and `completeOperationSuccess()`

### PR DESCRIPTION
**Pull request by tadgh**
Closes #2648 
* Add an allowlist for `$meta` operation and `/metadata` which skips running the consent interceptor for most checks. 
* still call `completeOperationSuccess` and `completeOperationFailure`.
* Add test
* Add some asserts to close out #2614

### CODE REVIEW

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentInterceptor.java [hunk 1]
```diff
@@ -330,4 +342,16 @@ private static ForbiddenOperationException toForbiddenOperationException(Consent
...
+	private boolean isAllowListedRequest(RequestDetails theRequestDetails) {
+		return isMetadataPath(theRequestDetails) || isMetaOperation(theRequestDetails);
+	}
+
+	private boolean isMetaOperation(RequestDetails theRequestDetails) {
+		return "$meta".equals(theRequestDetails.getOperation());
+	}
+
+	private boolean isMetadataPath(RequestDetails theRequestDetails) {
+		return "metadata".equals(theRequestDetails.getRequestPath());
```
Comment (fil512) [inline hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentInterceptor.java:355]: Use Constants.URL_TOKEN_METADATA

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentInterceptor.java [hunk 2]
```diff
@@ -330,4 +342,16 @@ private static ForbiddenOperationException toForbiddenOperationException(Consent
 		}
 		return new ForbiddenOperationException("Rejected by consent service", operationOutcome);
 	}
+
+	private boolean isAllowListedRequest(RequestDetails theRequestDetails) {
+		return isMetadataPath(theRequestDetails) || isMetaOperation(theRequestDetails);
+	}
+
+	private boolean isMetaOperation(RequestDetails theRequestDetails) {
+		return "$meta".equals(theRequestDetails.getOperation());
```
Comment (fil512) [inline hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentInterceptor.java:351]: Use JpaConstants.OPERATION_META

**Review by fil512 — APPROVED** [review]

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentInterceptor.java (hunk 2 above)
Comment (tadgh) [reply hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/consent/ConsentInterceptor.java:351]: Can't, not in the JPA package

**Comment by tadgh** [comment]
Closed in favour of https://github.com/hapifhir/hapi-fhir/pull/2649

**Pull request closed without merging**